frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/ReverificationResultHandlerTest.java

```diff
@@ -0,0 +1,170 @@
+            throws ParseException, UnsuccessfulReverificationResponseException {
+        when(reverificationResultService.getToken(any()))
+                .thenReturn(getUnsuccessfulTokenResponse());
+        when(reverificationResultService.sendIpvReverificationRequest(any()))
+                .thenThrow(
+                        new UnsuccessfulReverificationResponseException(
+                                "Error getting reverification result"));
+
+        var result = handler.handleRequest(apiRequestEventWithEmail("1234", EMAIL), context);
+
+        assertThat(result, hasStatus(400));
+        assertThat(result, hasJsonBody(ERROR_1058));
+    }
+
+    @Test
+    void shouldHandleIPVReverificationError()
+            throws ParseException, UnsuccessfulReverificationResponseException {
+        when(reverificationResultService.getToken(any())).thenReturn(getSuccessfulTokenResponse());
+        when(reverificationResultService.sendIpvReverificationRequest(any()))
+                .thenThrow(
+                        new UnsuccessfulReverificationResponseException(
+                                "Error getting reverification result"));
+
+        var result = handler.handleRequest(apiRequestEventWithEmail("1234", EMAIL), context);
+
+        assertThat(result, hasStatus(400));
+        assertThat(result, hasJsonBody(ERROR_1059));
+    }
+
+    private Map<String, String> getHeaders() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Session-Id", SESSION_ID);
+        headers.put("di-persistent-session-id", DI_PERSISTENT_SESSION_ID);
+        headers.put("X-Forwarded-For", IP_ADDRESS);
+        return headers;
+    }
+
+    public TokenResponse getSuccessfulTokenResponse() throws ParseException {
+        var tokenResponseContent =
+                "{"
+                        + "  \"access_token\": \"740e5834-3a29-46b4-9a6f-16142fde533a\","
+                        + "  \"token_type\": \"bearer\","
+                        + "  \"expires_in\": \"3600\""
+                        + "}";
+        var tokenHTTPResponse = new HTTPResponse(200);
+        tokenHTTPResponse.setEntityContentType(APPLICATION_JSON);
+        tokenHTTPResponse.setContent(tokenResponseContent);
+
+        return TokenResponse.parse(tokenHTTPResponse);
+    }
+
+    public TokenResponse getUnsuccessfulTokenResponse() throws ParseException {
+        var tokenResponseContent =
+                "{"
+                        + "\"error\": \"invalid_request\","
+                        + "\"error_description\": \"Request was missing the 'redirect_uri' parameter.\""
+                        + "}";
+
+        var tokenHTTPResponse = new HTTPResponse(400);
+        tokenHTTPResponse.setEntityContentType(APPLICATION_JSON);
+        tokenHTTPResponse.setContent(tokenResponseContent);
+
+        return TokenErrorResponse.parse(tokenHTTPResponse);
+    }
+
+    private APIGatewayProxyRequestEvent apiRequestEventWithEmail(String code, String email) {
```

Comment:
We've added a helper for an api gateway proxy request event, so this can become:

```
        var body = format("{ \"code\": \"%s\" , \"email\": \"%s\"}", code, email);
        return apiRequestEventWithHeadersAndBody(VALID_HEADERS, body);
```